File: app/src/main/java/com/tebet/mojual/view/message/MessageFragmentProvider.java
package com.tebet.mojual.view.message;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MessageFragmentProvider {

    @ContributesAndroidInjector(modules = MessageFragmentModule.class)
    abstract MessageFragment provideMessageFragmentFactory();
}

File: app/src/main/java/com/tebet/mojual/view/message/MessageFragmentProvider.kt
package com.tebet.mojual.view.message

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class MessageFragmentProvider {

    @ContributesAndroidInjector(modules = [MessageFragmentModule::class])
    internal abstract fun provideMessageFragmentFactory(): MessageFragment
}
